FWCore/ServiceRegistry/interface/ActivityRegistry.h

```diff
@@ -327,6 +327,18 @@ namespace edm {
     }
     AR_WATCH_USING_METHOD_1(watchPostEndProcessBlock)
 
+    typedef signalslot::Signal<void()> BeginProcessing;
+    /// signal is emitted just before the transitions from the Source will begin to be processed
+    BeginProcessing beginProcessingSignal_;
+    void watchBeginProcessing(BeginProcessing::slot_type const& iSlot) { beginProcessingSignal_.connect(iSlot); }
+    AR_WATCH_USING_METHOD_0(watchBeginProcessing)
+
+    typedef signalslot::Signal<void()> EndProcessing;
+    /// signal is emitted after all work has been done processing all transitions
```

Comment:
I suppose "all transitions" includes endProcessBlock, but not endJob(+endStream). How about clarifying along "all transitions except endJob and endStream"?

(otherwise I think the separate signal is a good clarification)

Reply:
changed to `processing all source transitions`